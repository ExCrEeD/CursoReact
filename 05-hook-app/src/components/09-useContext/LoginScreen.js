import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext);
    const usr = {
        id: 123,
        name:'daniel'
    }
    return (
        <div>
            <h1>login screen</h1>
            <button
                className="btn btn-primary"
                onClick={()=> setUser(usr)}
            >
                Login
            </button>
        </div>
    )
}
