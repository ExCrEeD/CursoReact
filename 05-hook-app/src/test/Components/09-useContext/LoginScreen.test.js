import { mount } from "enzyme";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('pruebas en <LoginScreen/>', () => {

    const setUser = jest.fn();
    const wrapper  = mount(
        <UserContext.Provider value={{setUser:setUser}}>
            <LoginScreen/>
        </UserContext.Provider>);
    test('debe de renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de ejecutar el setUser con el argumento esperado', () => {
       wrapper.find('button').prop('onClick')();
       expect(setUser).toHaveBeenCalledWith({
        id: 123,
        name:'daniel'
    });

    });
    
})
