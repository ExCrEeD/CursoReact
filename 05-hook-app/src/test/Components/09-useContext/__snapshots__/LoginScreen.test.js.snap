// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <LoginScreen/> debe de renderizar el componente 1`] = `
<div>
  <h1>
    login screen
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
